app/org/sagebionetworks/bridge/play/controllers/ReportController.java

```diff
@@ -254,6 +276,53 @@ public Result deleteStudyReportRecord(String identifier, String dateString) {
         return okResult("Report record deleted.");
     }
     
+    /**
+     * Update a single study report index. 
+     */
+    public Result updateStudyReportIndex(String identifier) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        
+        ReportIndex index = parseJson(request(), ReportIndex.class);
+        ReportDataKey key = new ReportDataKey.Builder()
+                .withHealthCode(session.getHealthCode())
+                .withReportType(ReportType.STUDY)
+                .withIdentifier(identifier)
+                .withStudyIdentifier(session.getStudyIdentifier()).build();
+        index.setKey(key.getIndexKeyString());
+        index.setIdentifier(identifier);
+        
+        reportService.updateReportIndex(ReportType.STUDY, index);
+        
+        return okResult("Report index updated.");
+    }
+    
+    /**
+     * Update a single participant report index. 
+     */
+    public Result updateParticipantReportIndex(String identifier) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        
+        ReportIndex index = parseJson(request(), ReportIndex.class);
+        ReportDataKey key = new ReportDataKey.Builder()
+                .withHealthCode(session.getHealthCode())
+                .withReportType(ReportType.PARTICIPANT)
+                .withIdentifier(identifier)
+                .withStudyIdentifier(session.getStudyIdentifier()).build();
+        index.setKey(key.getIndexKeyString());
+        index.setIdentifier(identifier);
+        
+        reportService.updateReportIndex(ReportType.PARTICIPANT, index);
+        
+        return okResult("Report index updated.");
+    }
+    
+    private void verifyIndex(final StudyIdentifier studyId, final String identifier) {
+        reportService.getReportIndices(studyId, ReportType.STUDY).getItems().stream()
+            .filter(index -> index.isPublic() && index.getIdentifier().equals(identifier))
+            .findFirst()
+            .orElseThrow(() -> new EntityNotFoundException(ReportIndex.class));
+    }
```

Comment:
This seems incredibly inefficient, especially given that the identifier is already a range key on the ReportIndex table. We can get the ReportIndex using a simple hash-and-range key query.

Also, the findFirst() is redundant, since there will only ever be at most one ReportIndex with that specific study, report type, and identifier combination.